### ISSUE TITLE
links deSerialize problem

### ISSUE THREAD

**Issue by gen4sp** [projectstorm/react-diagrams#40]
I've faced a problem. While deserializing simple diagram with custom nodes and ports this line returns null. But if I call it in a few msec - it returns correct value. 
It seems the canvas doesn't rendered yet when this code is run. 

Where could be a problem? 

diagramEngine.js - getNodePortElement()

```
var selector = this.canvas.querySelector('.port[data-name="' + port.getName() + '"][data-nodeid="' + port.getParent().getID()+'"]');
```

**Comment by dylanvorster** [comment]
does this still happen in the 2.4.8?

**Comment by gen4sp** [comment]
Yes, it throws an error now "Cannot find Node Port element with nodeID:..." 
It looks like just fired before the model was serialized. 
I am making a request to a server and trying to serialize the model. 

how code looks like
```
componentDidMount() {

    API.httpGet('/stage')
      .then((stageJson) => {
        let model2 = new SRD.DiagramModel();
        this.state.engine.setDiagramModel(model2);
        model2.deSerializeDiagram(stageJson, this.state.engine);
        subscribeModel(model2); //subscribe to model events
        this.setState({model:model2});
        _.each(model2.getNodes(),subscribeNode.bind(this)); //subscribe to node events
        this.forceUpdate()
      });
  }
```

**Comment by brunoalano** [comment]
I had the same error. I created a new `LinkWidget` and `LinkFactory`. I'm searching where can be the problem.

**Comment by brunoalano** [comment]
I a workaround that I made is reimporting 2 times, with a timeout of 100 milliseconds.

```

**Comment by ferdaarikan** [comment]
I have a similar issue. I am using default factories. When I deserialize the diagram with a new engine and empty model links don't get created with an error like 

> Cannot find Node Port element with nodeID: [1e0a6c3f-592b-49ab-a953-0b8eb67e1a55] and name: [sensorValue]

If I deserialize the model twice links appear. Probably something to do with deserialization order. Nodes need to be created before the links are created. I can see this in the logs as well, first links are added and then removed due to the node not found error message.

**Comment by dylanvorster** [comment]
arghhh race conditions :C If you can provide me with a test case ill try fix it

**Comment by ferdaarikan** [comment]
Hi, Thanks for looking into this. I have created a small [example](https://github.com/ferdaarikan/storm-react-diagrams-example). When app is running clicking save, clear, load buttons reproduces the issue.

**Comment by dylanvorster** [comment]
Thanks for taking the time to do this, Ill jump onto this shortly and see what I can do.

**Comment by sanfilippopablo** [comment]
Any update on this?

**Comment by sanfilippopablo** [comment]
Hold on, I think I got it. I'll send a PR in a few minutes.

**Comment by sanfilippopablo** [comment]
I think this is fixed in PR #80 , can anyone confirm?

**Comment by dylanvorster** [comment]
Fixed in caadbf4a9a0b03bde4ee32c5f1b9789ee7ed34bb

**Comment by dylanvorster** [comment]
Fixed in 2.5.2
